README.md

```diff
@@ -1,2 +1,90 @@
-# base-visualisation-builder
-A base project for visualisation builders on DAFNI. This project has in it the code needed to authenticate and authorise users on DAFNI which will allow you to focus on the code to actually implement the visualisation builder you want to create.
+# Base Visualisation Builder
+
+A base project for visualisation builders on DAFNI. This project has in it the code needed to
+authenticate and authorise users on DAFNI which will allow you to focus on the code to actually
+implement the visualisation builder you want to create.
+
+The project uses [Vue](https://v2.vuejs.org/) and [Vuetify](https://vuetifyjs.com/en/) as these
+are the tools DAFNI uses to build its Front Ends. The only restrictions visualisation builders on
+DAFNI have are; firstly they must be built as web applications so that they can be deployed and
+loaded in a browser, secondly they must be linked into DAFNI's Keycloak auth service otherwise
+we will not be able to provide your visualisation with any data. If you would prefer to write
+your visualisation builder in React/Angular or even pure HTML + JS then that is up to you and it
+should work on DAFNI, this base project is provided purely for your convenience as a quick way to
+start writing your builder without needing to worry about how to login to Keycloak, get the
+visualisation instance details or data files from DAFNI etc.
+
+Below we will briefly go over each folder, the contents and whether they are necessary for your visualisation builder to work on DAFNI.
+
+## api
+
+This folder contains the code that calls across to DAFNI's APIs. There are functions to call to get a visualisation instance, get the list of files contained in a list of dataset versions and download the a file from our Minio store. The backends folder inside `api` contains a script that will load the API urls into the application so that they can be called by your code.
+
+The entirety of this folder is required for your visualisation builder to work correctly.
+
+## assets
+
+This folder contains the same stylesheets that are used to create the DAFNI facility and Documentation sites. We make these available to allow you to match your visualisation builder to DAFNI's style if you would like to, though that is up to you.
+
+None of this folder is required for your visualisation builder to work correctly. Feel free to delete it or change some of the contents. NOTE - if you do delete the folder then you may need to change some other files that currently rely on the contents of this folder.
+
+## backends-templates
+
+This folder contains some template JSON files that when deployed will be filled with DAFNI's API urls.
+
+This folder is required for your visualisation builder to work correctly. You should be able to just ignore this folder completely.
+
+## components
```

Comment:
If I'm right, this dir and a few others are part of the standard Vue framework. Because of this they can be 
a) Removed if you know what you're doing with Vue 
b) Not removed because Vue needs them.

Could we make that more clear somehow? Perhaps put 'Standard Vue Directory' at the start of the description? I would suggest putting all the Vue stuff at the end, but I do like that they're in alphabetical order.

Clarifying it means that people who decide not to use Vue, can quickly filter these out and focus their attention on the bits that are there because DAFNI needs them (rather than because Vue needs).

Reply:
Depending on the Vue directory they can either be a) or b) yes, at least in a standard Vue project. But there are some of those folders Vue doesn't require this base project does require.

I have added that flag to the headings for each of the folders. 

People who don't use Vue may still need to look into those folders because there is DAFNI specific code in them. I would not recommend someone try create an app that logs in to DAFNI while avoiding all folders flagged with `Standard Vue Directory`.